tests/provider/test_no_op_provider.py

```diff
@@ -0,0 +1,68 @@
+from numbers import Number
+
+from src.open_feature_api import OpenFeatureAPI
+from src.provider.no_op_provider import NoOpProvider
+
+
+def setup():
+    open_feature = OpenFeatureAPI
+    open_feature.set_provider(NoOpProvider())
+    provider = open_feature.get_provider()
+    assert isinstance(provider, NoOpProvider)
+
+
+def test_should_get_boolean_flag_from_no_op():
+    # Given
+    open_feature = OpenFeatureAPI()
```

Comment:
Seems odd that you're using an instance here to call the static method. I had to go look twice to see if we were hitting the OFAPI = global singleton requirement. Not sure how that's typically done in python

Reply:
Understood, thanks @toddbaert & @justinabrahms. 

In the latest commit, we've ensured that the API itself and the provider are singletons, just in slightly different ways. 

The usage from a client perspective should look something like: 

```python
import open_feature_api

open_feature_api.set_provider(NoOpProvider())
client = open_feature_api.get_client()
```

There is an alternative that uses an `OpenFeatureAPI` class to keep it more in key with the Java implementation, however, this module / function based implementation is definitely the most pythonic way of doing it.

Reply:
Please don't make it java-y. ;)